### PR TITLE
Reporting errors via return codes when a node that is not present in any jobs is unpublished.

### PR DESCRIPTION
**Pull request by karok2m**
Azure DevOps PBI: [[REST API] unpublishing an unknown node, doesn't indicate an error](https://msazure.visualstudio.com/One/_workitems/edit/9056770)

Changes:
* `PublisherJobService.NodePublishStopAsync()` method has been modified to throw an exception when a node that is not present in any jobs is unpublished.
* `PublisherJobService.NodePublishBulkAsync()` method has been modified to report `404/NotFound` status code for unpublish node requests for nodes that are not present in any jobs.
* `PublishController` controller class has been modified to report `404/NotFound` status codes when appropriate.
* `DefaultJobService.DeleteJobAsync()` method has been modified to throw an exception when it is called with an unknown `jobId`.
* Added `CancellationToken` to `IPublishServices` interface.

### CODE REVIEW

File: api/src/Microsoft.Azure.IIoT.OpcUa.Api.Publisher/src/Models/PublishBulkResponseApiModel.cs [hunk 1]
```diff
@@ -18,13 +18,13 @@ public class PublishBulkResponseApiModel {
         /// </summary>
         [DataMember(Name = "nodesToAdd", Order = 0,
             EmitDefaultValue = false)]
-        public List<ServiceResultApiModel> NodesToAdd { get; set; }
+        public Dictionary<string, ServiceResultApiModel> NodesToAdd { get; set; }
```
Comment (koepalex) [inline api/src/Microsoft.Azure.IIoT.OpcUa.Api.Publisher/src/Models/PublishBulkResponseApiModel.cs:21]: Please use IDictionary

File: api/src/Microsoft.Azure.IIoT.OpcUa.Api.Publisher/src/Models/PublishBulkResponseApiModel.cs [hunk 2]
```diff
@@ -18,13 +18,13 @@ public class PublishBulkResponseApiModel {
...
 
         /// <summary>
         /// Node to remove
         /// </summary>
         [DataMember(Name = "nodesToRemove", Order = 1,
             EmitDefaultValue = false)]
-        public List<ServiceResultApiModel> NodesToRemove { get; set; }
+        public Dictionary<string, ServiceResultApiModel> NodesToRemove { get; set; }
```
Comment (koepalex) [inline api/src/Microsoft.Azure.IIoT.OpcUa.Api.Publisher/src/Models/PublishBulkResponseApiModel.cs:28]: Please use IDictionary

**Review by koepalex — APPROVED** [review]
:shipit:

File: api/src/Microsoft.Azure.IIoT.OpcUa.Api.Publisher/src/Models/PublishBulkResponseApiModel.cs (hunk 1 above)
Comment (karok2m) [reply api/src/Microsoft.Azure.IIoT.OpcUa.Api.Publisher/src/Models/PublishBulkResponseApiModel.cs]: Fixed.

**Review by koepalex — APPROVED** [review]
:shipit:

**Review by barnstee — APPROVED** [review]

**Pull request merged**